src/components/invitation/ItemOptionModal.tsx

```diff
@@ -77,13 +83,27 @@ const ItemOptionModal: FunctionComponent<ItemOptionModalProps> = ({
       {}
     ) as OptionDictOptionData;
     let haveAllMandatoryOption = true;
-    listify(dishOption, (_, value) => ({ ...value })).forEach((menuOption) => {
+    listify(dishOption, (key, value) => ({ optionId: key, ...value })).forEach((menuOption) => {
```

Comment:
Technically optionId was already in `value` please check dishOption again to verify

Reply:
Fixed